File: cobaco-server-spring/src/main/java/dentaira/cobaco/server/file/FileRepository.java
package dentaira.cobaco.server.file;

import java.io.InputStream;
import java.util.List;
import java.util.UUID;

public interface FileRepository {

    public UUID generateId();

    public List<StoredFile> searchRoot(Owner owner);

    public List<StoredFile> search(String parentDirId, Owner owner);

    public StoredFile findById(String id, Owner owner);

    public InputStream findContentById(String id, Owner owner);

    public List<StoredFile> searchForAncestors(StoredFile file);

    public void save(StoredFile file);

    public void delete(StoredFile file);
}

File: cobaco-server-play/app/dentaira/cobaco/server/file/FileRepository.scala
package dentaira.cobaco.server.file

import scalikejdbc.{AutoSession, DBSession}

import java.util.UUID

trait FileRepository {

  def generateId(): UUID

  def findById(id: String)(implicit session: DBSession = AutoSession): Option[StoredFile]

  def save(storedFile: StoredFile)(implicit session: DBSession = AutoSession)

  def delete(id: String)(implicit session: DBSession = AutoSession)

  def update(file: StoredFile)(implicit session: DBSession = AutoSession)

}
